kong/db/dao/plugins/go.lua

```diff
@@ -36,6 +36,47 @@ do
   end
 end
 
+do
+  local pluginserver_proc
+
+  function go.manage_pluginserver()
+    if pluginserver_proc then
+      -- already being managed
+      return
+    end
+
+    if ngx.worker.id() ~= 0 then
+      -- only one manager
+      pluginserver_proc = true
+      return
+    end
+
+    ngx_timer_at(0, function ()
+      local ngx_pipe = require "ngx.pipe"
+
+      while true do
+        kong.log.notice("Starting go-pluginserver")
+        pluginserver_proc = assert(ngx_pipe.spawn({
+          kong.configuration.prefix .. "/bin/go-pluginserver",
```

Comment:
We should not assume the pluginserver will be available in the prefix; by default it will not (all our distributions place it under `/usr/local/bin`. Something like the following should do it:

```
pluginserver_proc = assert(ngx_pipe.spawn({
    "go-pluginserver",
    "-socket", go.socket_path(),
    "-plugins-directory", kong.configuration.go_plugins_dir,
   }, {
    environ = { "PATH=/usr/local/bin" }
   }))
```

Reply:
unfortunately, the `environ` variables apply only for the spawned process, not while searching the executable.  i'll hardcode the complete `/usr/local/bin/go-pluginserver` path.

Reply:
>  i'll hardcode the complete /usr/local/bin/go-pluginserver path.

This is bad, it will cause trouble with testing environments and such. If `ngx_pipe.spawn` doesn't know how to search the PATH we have to either do it ourselves or add a `kong.conf` option for the full path of the pluginserver, which kinda sucks. I would loop over the PATH variable searching for the file before spawning.

Reply:
the problem is that nginx cleans up the environment.  would have to add an `env PATH;` line to the config so `pipe.spawn()` can use it.

Reply:
Oh, I see. In that case I would go with a `kong.conf` option for the full path of the pluginserver. The default can be set to `/usr/local/bin/go-pluginserver` meaning that our shipped binaries work out-of-the-box, but it's still overridable in dev environments.

Reply:
done